compiler/luci/pass/src/FoldDequantizePass.cpp

```diff
@@ -74,31 +80,41 @@ luci::CircleConst *dequantized_const_node(luci::CircleConst *const_node)
   for (uint32_t i = q_dim + 1; i < const_node->rank(); ++i)
     right_count *= const_node->dim(i).value();
 
-  if (const_node->dtype() == loco::DataType::S8)
+  for (uint32_t i = 0; i < new_const_node->size<loco::DataType::FLOAT32>(); ++i)
   {
-    for (uint32_t i = 0; i < const_node->size<loco::DataType::S8>(); ++i)
-    {
-      uint32_t qd = (i % right_count) / (right_count / q_dim_value);
-      if (qd >= const_node->quantparam()->zerop.size())
-        qd = 0;
+    uint32_t qd = (i % right_count) / (right_count / q_dim_value);
+    if (qd >= const_node->quantparam()->zerop.size())
+      qd = 0;
 
-      new_const_node->at<loco::DataType::FLOAT32>(i) =
-        (float)(const_node->at<loco::DataType::S8>(i) - const_node->quantparam()->zerop.at(qd)) *
-        const_node->quantparam()->scale.at(qd);
-    }
-  }
-  else
-  {
-    for (uint32_t i = 0; i < const_node->size<loco::DataType::U8>(); ++i)
+    switch (const_node->dtype())
     {
-      uint32_t qd = (i % right_count) / (right_count / q_dim_value);
-      if (qd >= const_node->quantparam()->zerop.size())
-        qd = 0;
-
-      new_const_node->at<loco::DataType::FLOAT32>(i) =
-        (float)((int)const_node->at<loco::DataType::U8>(i) -
-                const_node->quantparam()->zerop.at(qd)) *
-        const_node->quantparam()->scale.at(qd);
+      case loco::DataType::S8:
+        new_const_node->at<loco::DataType::FLOAT32>(i) =
+          (float)(const_node->at<loco::DataType::S8>(i) - const_node->quantparam()->zerop.at(qd)) *
+          const_node->quantparam()->scale.at(qd);
+        break;
+      case loco::DataType::S16:
+        new_const_node->at<loco::DataType::FLOAT32>(i) =
+          (float)(const_node->at<loco::DataType::S16>(i) - const_node->quantparam()->zerop.at(qd)) *
+          const_node->quantparam()->scale.at(qd);
+        break;
+      case loco::DataType::S32:
+        new_const_node->at<loco::DataType::FLOAT32>(i) =
+          (float)(const_node->at<loco::DataType::S32>(i) - const_node->quantparam()->zerop.at(qd)) *
+          const_node->quantparam()->scale.at(qd);
+        break;
+      case loco::DataType::S64:
+        new_const_node->at<loco::DataType::FLOAT32>(i) =
+          (float)(const_node->at<loco::DataType::S64>(i) - const_node->quantparam()->zerop.at(qd)) *
```

Comment:
You should use `static_cast` as this is safer way in c++. And, is it safe to convert s64 to float? It becomes to have narrower bits.

Reply:
> You should use static_cast as this is safer way in c++

I will apply this :)

> is it safe to convert s64 to float? It becomes to have narrower bits.

"Safe" in your saying is about some disaster like "2^63-1(int64_t) --> static_cast --> 2^32(float)"?
If so, it is safe because the range of `float` is wider than `int64_t` :)
```
Float Range : [-3.40282e+38, 3.40282e+38]
int64_t Range : [-9223372036854775808, 9223372036854775807]
```

Of course, there are inevitable error caused by conversion between integer and float.
But your worrying may not occur :D

Reply:
Well, floating point system is not like one to one mapping. Since 23bit(float mantissa) can be one to one mapping, bigger integer may lose precision. But, seems to me that the data are lower than 2^23. So, it will be okay.

Reply:
> floating point system is not like one to one mapping.

This is what I thought as "inevitable error caused by conversion between integer and float" :)
So even we use `double` instead of `float`, the problem itself is not removed T.T
The most important thing is... because of `new_const_node->at<loco::DataType::FLOAT32>`, the final result is always float even we use `double` ^.T

Reply:
Yes. Actually, it doesn't needed to be perfectly resolved because integer bigger than 2^32 may not come in.  It's better it prints some warning if it happens to come in:)